remoteworkitem/trackeritem_repository.go

```diff
@@ -5,6 +5,14 @@ import "github.com/jinzhu/gorm"
 // upload imports the items into database
 func upload(db *gorm.DB, tqID int, item map[string]string) error {
 	ti := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
-	err := db.Create(&ti).Error
+	db.Where("remote_item_id = ?", item["id"]).Find(&ti)
+	var err error
+	if ti.ID == 0 {
+		ti1 := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
+		err = db.Create(&ti1).Error
+	} else {
+		ti2 := TrackerItem{ID: ti.ID, Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
```

Comment:
@baijum Ah, no, you still need to check for the tracker_query even if remote_id is unique. And remote_id can't be unique. :)

TrackerQuery will define e.g. which WorkItemType the found remote issues should map to. Different users could remote import the same issues but map them differently.